dotnet/src/SemanticKernel.Core/Data/TextSearch/VectorStoreRecordTextSearch.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Embeddings;
+
+namespace Microsoft.SemanticKernel.Data;
+
+/// <summary>
+/// A Vector Store Text Search implementation that can be used to perform searches using a <see cref="IVectorStoreRecordCollection{TKey, TRecord}"/>.
+/// </summary>
+[Experimental("SKEXP0001")]
+public sealed class VectorStoreRecordTextSearch<TRecord> : ITextSearch
```

Comment:
In terms of the naming: `VectorStoreRecordTextSearch`.  Do you envisage other types of VectorStore TextSearchers to this one?  Just wondering about the Record part.